server/src/repositories/reportRepository.js

```diff
@@ -8,9 +8,14 @@ const reportRepository = (repository) => {
     return repository.productsCount(callback);
   }
 
+  function aggregateByProductId(id, callback) {
+    return repository.aggregateByProductId(id, callback);
+  }
+
   return {
     add,
     aggregateByProduct,
+    aggregateByProductId,
```

Comment:
Why do aggregateByProduct and aggregateByProductId have different logic? I would expect them to do the same, the only difference being the expected argument (product vs product id). If we want them to act differently, we should name them appropriately - so that the naming describes what the method does.

Reply:
I changed those names to:
aggregateByProduct => aggregateCompatibilityByProduct
aggregateByProductId => aggregateBBDetailsByProductId